docs/rfc/rfc-010-p2p-light-client.rst

```diff
@@ -0,0 +1,127 @@
+==================================
+RFC 010: Peer to Peer Light Client
+==================================
+
+Changelog
+---------
+
+- 2022-01-21: Initial draft (@tychoish)
+
+Abstract
+--------
+
+The dependency on access to the RPC system makes running or using the light
+client more complicated than it should be, because it is difficult to get
+access to the RPC endpoints on a running node. There is no deep dependency for
```

Comment:
It is worth pointing out that this is not a consequence of RPC per se, but because many node operators firewall or block it.

That, in turn, is because the RPC service exposes a very large API, some of which would be fine to share but much of which is not.

While that's only tangential to the light client interest, I think it helps motivate the move to P2P, which _always_ has to be on for a node to exist.

(I would also like to narrow and make RPC more selective, but that's orthogonal)

Reply:
I agree—I don't intend a case against it, rather to highlight the pain points that are specifically circumstantial to RPC vs. the intrinsic advantages of P2P.